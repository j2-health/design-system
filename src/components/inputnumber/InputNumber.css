.j2-input-number {
  &.ant-input-number {
    &:hover {
      .ant-input-number-handler-wrap {
        max-width: 100%;
        width: auto;
      }
    }

    .ant-input-number-handler-wrap {
      max-width: 0;
      transition: max-width 0.2s;
      .ant-input-number-handler {
        height: 50%;
        padding: 0 12px;
        color: var(--j2-color-primary);
      }

      .ant-input-number-handler-up {
        align-items: flex-end;
      }

      .ant-input-number-handler-down {
        border-block-start: none;
        align-items: flex-start;
      }
    }
  }
}
