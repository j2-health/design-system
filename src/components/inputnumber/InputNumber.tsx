import { InputNumber as AntdInputNumber, InputNumberProps } from 'antd'
import cx from 'classnames'
import './InputNumber.css'
import { CaretDown, CaretUp } from '@phosphor-icons/react'
type Props = Expand<InputNumberProps>

export const InputNumber = ({ size = 'large', ...props }: Props) => {
  return (
    <AntdInputNumber
      className={cx('j2-input-number', props.className)}
      upHandler={<CaretUp weight="bold" size={12} />}
      downHandler={<CaretDown weight="bold" size={12} />}
      size={size}
      {...props}
    />
  )
}
