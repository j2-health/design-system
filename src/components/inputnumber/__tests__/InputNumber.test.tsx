import { render } from '@testing-library/react'
import { InputNumber } from '../InputNumber'

describe('InputNumber', () => {
  it('should render correctly', () => {
    const { container } = render(<InputNumber />)
    expect(container).toMatchSnapshot()
  })
})
