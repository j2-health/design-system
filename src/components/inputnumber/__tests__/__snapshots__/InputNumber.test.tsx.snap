// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputNumber should render correctly 1`] = `
<div>
  <div
    class="ant-input-number ant-input-number-lg css-dev-only-do-not-override-tpassh ant-input-number-outlined j2-input-number"
  >
    <div
      class="ant-input-number-handler-wrap"
    >
      <span
        aria-disabled="false"
        aria-label="Increase Value"
        class="ant-input-number-handler ant-input-number-handler-up"
        role="button"
        unselectable="on"
      >
        <svg
          fill="currentColor"
          height="12"
          viewBox="0 0 256 256"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M216.49,168.49a12,12,0,0,1-17,0L128,97,56.49,168.49a12,12,0,0,1-17-17l80-80a12,12,0,0,1,17,0l80,80A12,12,0,0,1,216.49,168.49Z"
          />
        </svg>
      </span>
      <span
        aria-disabled="false"
        aria-label="Decrease Value"
        class="ant-input-number-handler ant-input-number-handler-down"
        role="button"
        unselectable="on"
      >
        <svg
          fill="currentColor"
          height="12"
          viewBox="0 0 256 256"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M216.49,104.49l-80,80a12,12,0,0,1-17,0l-80-80a12,12,0,0,1,17-17L128,159l71.51-71.52a12,12,0,0,1,17,17Z"
          />
        </svg>
      </span>
    </div>
    <div
      class="ant-input-number-input-wrap"
    >
      <input
        autocomplete="off"
        class="ant-input-number-input"
        role="spinbutton"
        step="1"
        value=""
      />
    </div>
  </div>
</div>
`;
